### PR TITLE
Semihosting: Add support of SYS_READC

### PR DESCRIPTION
**Pull request by Fabien-Chouteau**

### CODE REVIEW

File: src/gdbserver/semihosting.c [hunk 1]
```diff
@@ -457,6 +457,13 @@ int do_semihosting (stlink_t *sl, uint32_t r0, uint32_t r1, uint32_t *ret) {
         }
         break;
     }
+    case SYS_READC:
+    {
+        uint8_t c = getchar();
+        *ret = c;
+        return 0;
+        break;
```
Comment (potozi) [inline src/gdbserver/semihosting.c]: will be never executed

Comment (xor-gate) [inline src/gdbserver/semihosting.c]: Yes, break is unnessary here.

Comment (Fabien-Chouteau) [inline src/gdbserver/semihosting.c]: And that's why you don't push a commit Friday evening... :confused:

**Comment by Fabien-Chouteau** [comment]
Updated.

**Comment by xor-gate** [comment]
Thanks!

**Pull request merged**